### ISSUE TITLE
Empty directories in build context are ignored

### ISSUE THREAD

**Issue by MortlMcCrisis** [bmuschko/gradle-docker-plugin#429]
Hello everybody,

I am trying to build an image with a file structure which contains empty directories. When using docker the image is built as expected. When using gradle and the gradle-docker-plugin the empty directories are omitted and not present in the resulting image.

I did not found a parameter to control this behavior. I also did not find a bug report, either for gradle-docker-plugin nor for docker-java which explains this behavior.

The filestructure looks as following:

Dockerfile
server
\- emptyDir
\- nonEmptyDir
--file.txt

Dockerfile:
```
FROM debian:jessie
ADD server /opt/server
```

build.gradle:
```
import com.bmuschko.gradle.docker.tasks.container.*
import com.bmuschko.gradle.docker.tasks.image.*

buildscript {
	dependencies {
		classpath 'com.bmuschko:gradle-docker-plugin:3.0.12'
	}
}

plugins {
  id 'com.bmuschko.vagrant' version '2.0'
}

apply plugin: 'com.bmuschko.docker-remote-api'

docker {
	url = 'http://xpro-jslave01.srv.inxmail.de:4243'
}

task buildServerDockerImage(type: DockerBuildImage) {
	inputDir = project.file("docker")
	tag = "test/testserver:1.0.0"
}
```

I also created a minimal example project to reproducte this issue: [test.zip](https://github.com/bmuschko/gradle-docker-plugin/files/1200313/test.zip)

When you run Docker from within the docker-Folder to build an image (empty directories are present):
```
docker build -t test/testserver:1.0.0 .
docker run -it test/testserver:1.0.0 bash

cd /opt/server
ls
```

When you use gradle and the gradle-docker-plugin (empty directories are missing):
```
gradlew buildServerDockerImage
docker run -it test/testserver:1.0.0 bash

cd /opt/server
ls
```

Does anybody know what is triggering this issue?

**Comment by cdancy** [comment]
Indeed I was able to reproduce. Think this may be relevant to this issue:

https://github.com/bmuschko/gradle-docker-plugin/issues/187

@orzeh any thoughts here? You could always create those empty directories of course but having something automated and just work like you would expect would indeed be better.

**Comment by orzeh** [comment]
@MortlMcCrisis @cdancy I suspect it is related to how `docker-java` packs build context into `tar` archive. I'll take a look at it in spare time.

**Comment by cdancy** [comment]
Yeah @MortlMcCrisis you may want to ask the docket-java guys about this.

**Comment by orzeh** [comment]
@MortlMcCrisis @cdancy it's related to `docker-java`, please track [this PR](https://github.com/docker-java/docker-java/pull/889).

**Comment by cdancy** [comment]
Thanks @orzeh !!!

**Comment by MortlMcCrisis** [comment]
Thank you very much @orzeh and @cdancy !
For the moment I will try to make a fork and implement the fix manually, because I can not wait for the next Version of docker-java and gradle-docker-plugin.
I will also try to help with the fix but unfortunately I think I can not do much more since the fix already exists.

**Comment by MortlMcCrisis** [comment]
Can you estimate when the next version with this fix will be released?

**Comment by cdancy** [comment]
@MortlMcCrisis commented on the ISSUE to see if we can get this fast-tracked and in. Can't promise a timeline but those guys are usually pretty good in helping out and releasing when we need something.

**Comment by cdancy** [comment]
@MortlMcCrisis released version `3.1.0` which should address this issue. Closing this issue out but if it doesn't work feel free to re-open.

**Comment by MortlMcCrisis** [comment]
Thank you very much! I am using the new version, got rid of my workaround and it works as expected now.
